go/cmd/dolt/commands/push.go

```diff
@@ -88,6 +89,53 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, pushDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
+	if apr.Contains(cli.AllFlag) {
+		return cmd.pushAllBranches(ctx, apr, dEnv, usage)
+	}
+
+	return cmd.doPush(ctx, apr, dEnv, usage)
+}
+
+func (cmd PushCmd) pushAllBranches(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.DoltEnv, usage cli.UsagePrinter) int {
+	branches, err := dEnv.DoltDB.GetBranches(ctx)
```

Comment:
This includes e.g. remote branches too, right? Or are those stored under a different ref namespace?